From c655360c8297b65be27b8a4da6f9541edfed80f9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=89tienne=20Noss?= <etienne.noss@wifirst.fr>
Date: Sat, 11 Mar 2017 16:36:01 +0100
Subject: [PATCH] tc u32: fix stats when there ara attached actions

The u32 filter returned the stats2 class instead of an instance of it,
making it crash when trying to parse the filter's associated action's
stats.
---
 pyroute2/netlink/rtnl/tcmsg/cls_u32.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/pyroute2/netlink/rtnl/tcmsg/cls_u32.py b/pyroute2/netlink/rtnl/tcmsg/cls_u32.py
index 2a4f8629e125..4dee5b791a60 100644
--- a/pyroute2/netlink/rtnl/tcmsg/cls_u32.py
+++ b/pyroute2/netlink/rtnl/tcmsg/cls_u32.py
@@ -110,10 +110,9 @@ class options(nla, nla_plus_police):
                        ('TCA_ACT_KIND', 'asciiz'),
                        ('TCA_ACT_OPTIONS', 'get_act_options'),
                        ('TCA_ACT_INDEX', 'hex'),
-                       ('TCA_ACT_STATS', 'get_stats2'))
+                       ('TCA_ACT_STATS', 'stats2'))
 
-            def get_stats2(self, *argv, **kwarg):
-                return stats2
+            stats2 = stats2
 
     class u32_sel(nla):
         fields = (('flags', 'B'),
-- 
2.11.0

